exam001/Views/ViewController.swift

```diff
@@ -0,0 +1,482 @@
+            ])
+    }
+    
+    // MARK: - tablview
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return fireBaseModel.getTableDatasize()
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = DevCell(style: UITableViewCell.CellStyle.value1, reuseIdentifier: "DevCell")
+        cell.cellIndex = indexPath.row
+        let singleCellData: NSDictionary = fireBaseModel.getTableData(indexPath.row)
+        let location = singleCellData[self.viewModel.fieldsArray[3]] as? String ?? ""
+        let stat = singleCellData["Status"] as? String ?? ""
+        cell.textLabel!.text = location
+        cell.detailTextLabel?.text = stat
+        cell.textLabel?.textColor = UIColor.orange
+        cell.backgroundColor = UIColor.darkGray
+        
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        print("Num: \(indexPath.row)")
+        print(fireBaseModel.tableCellDataArray[indexPath.row] as Any)
+        if let dictv = fireBaseModel.tableCellDataArray[indexPath.row] {
+            firstTextField.text = dictv[self.viewModel.fieldsArray[0]] as? String
+            secondTextField.text = dictv[self.viewModel.fieldsArray[1]] as? String
+            thirdTextField.text = dictv[self.viewModel.fieldsArray[2]] as? String
+            fourthTextField.text = dictv[self.viewModel.fieldsArray[3]] as? String
+        }
+        if let navigationController = self.navigationController {
+            print("T##items: Any...##Any")
+            print(indexPath.row)
+            print("T##items: Any...##Any")
+            let detailViewController = RequestDetailsViewController(fireBaseModel.tableCellDataArray[indexPath.row] as! NSDictionary,indexPath.row)
+            navigationController.pushViewController(detailViewController, animated: true)
+        }
+//        let vc = RequestDetailsViewController()
+//        self.present(vc, animated: true, completion: nil)
+
+    }
+    
+    // MARK: - Actions
+    @objc func topBtnTapped() {
+        UIView.animate(withDuration: 2.0, animations: {
+            self.subStack.removeArrangedSubview(self.secondTextField)
+            self.subStack.removeArrangedSubview(self.fourthTextField)
+            self.containerView.layoutIfNeeded()
+        })
+        
+    }
+    
+    @objc func bottomBtnTapped() {
+        UIView.animate(withDuration: 2.0, animations: {
+            self.subStack.insertArrangedSubview(self.secondTextField, at: 1)
+            self.subStack.insertArrangedSubview(self.fourthTextField, at: 3)
+            self.containerView.layoutIfNeeded()
+        })
+        
+    }
+    
+    @objc func submitBtnTapped() {
+        if (firstTextField.text?.isEmpty)! || (secondTextField.text?.isEmpty)! || (thirdTextField.text?.isEmpty)! || (fourthTextField.text?.isEmpty)! {
```

Comment:
`firstTextField.text?`
`text` might be `nil` which will lead to app crash, 
use `if let` or `guard let` to avoid crashes.